src/strategies/bearerTokenStrategy.ts

```diff
@@ -2,35 +2,69 @@
  * Builds and configure a Passport strategy to authenticate the proxy clients.
  */
 
+import * as express from "express";
 import { Either } from "fp-ts/lib/Either";
 import * as passport from "passport-http-bearer";
+import { IVerifyOptions } from "passport-http-bearer";
 import container, { SESSION_STORAGE } from "../container";
 import { ISessionState, ISessionStorage } from "../services/ISessionStorage";
-import { SessionToken } from "../types/token";
+import { SessionToken, WalletToken } from "../types/token";
 
 const bearerTokenStrategy = () => {
-  return new passport.Strategy((token, done) => {
+  const options = {
+    passReqToCallback: true,
+    realm: "Proxy API",
+    scope: "request"
+  };
+  return new passport.Strategy(options, (
+    req: express.Request,
+    token: string,
+    // tslint:disable-next-line:no-any
+    done: (error: any, user?: any, options?: IVerifyOptions | string) => void
+  ) => {
+    const path = req.route.path;
     const sessionStorage: ISessionStorage = container.resolve(SESSION_STORAGE);
 
-    sessionStorage.get(token as SessionToken).then(
-      (errorOrSessionState: Either<Error, ISessionState>) => {
-        errorOrSessionState.fold(
-          () => done(undefined, false),
-          sessionState => {
-            // Check if the session is expired.
-            if (sessionState.expireAt.getTime() < Date.now()) {
-              done(undefined, false);
-            } else {
-              done(undefined, sessionState.user);
-            }
-          }
-        );
-      },
-      () => {
-        done(undefined, false);
-      }
-    );
+    if (path.startsWith("/api")) {
```

Comment:
better to define constants for `/api` and `/pagopa/api` and make them very visible (instead of hardcoding the values here) - suppose we change the paths in the API specs, who will think about looking in this file? :)

even better would be to have these paths as part of the config (or perhaps even reading them from the yml specs?)